Amby/Util/Service/LoginService.swift

```diff
@@ -0,0 +1,76 @@
+//
+//  LoginService.swift
+//  Amby
+//
+//  Created by tenma.i on 2019/10/16.
+//  Copyright © 2019 eifandevs. All rights reserved.
+//
+
+import Firebase
+import Foundation
+import GoogleSignIn
+import Hydra
+
+class LoginService {
+    var isLoggedIn: Bool {
+        return Auth.auth().currentUser != nil
+    }
+
+    func signIn(_: GIDSignIn!, didSignInFor user: GIDGoogleUser!, withError error: Error!) -> Promise<Bool> {
+        return Promise<Bool>(in: .main, token: nil) { resolve, reject, _ in
+            if let error = error {
+                log.error("google signIn error. error: \(error.localizedDescription)")
+                reject(error)
+            } else {
+                // Perform any operations on signed in user here.
+                let userId = user.userID // For client-side use only!
+                let idToken = user.authentication.idToken // Safe to send to the server
+                let fullName = user.profile.name
+                let givenName = user.profile.givenName
+                let familyName = user.profile.familyName
+                let email = user.profile.email
+                log.debug("userId: \(userId ?? "") idToken: \(idToken) fullName: \(fullName) givenName: \(givenName) familyName: \(familyName) email: \(email)")
+                guard let authentication = user.authentication else { return }
+                let credential = GoogleAuthProvider.credential(withIDToken: authentication.idToken,
+                                                               accessToken: authentication.accessToken)
+                Auth.auth().signIn(with: credential) { _, error in
+                    if let error = error {
+                        log.error("firebase signIn error. error: \(error.localizedDescription)")
+                        reject(error)
+                        return
+                    }
+                    log.debug("firebase login success. userid: \(Auth.auth().currentUser!.uid)")
+                    resolve(true)
+                    // User is signed in
+                    // ...
+                }
+            }
+        }
+    }
+
+    func signOut() {
+        GIDSignIn.sharedInstance()!.signOut()
+        try! Auth.auth().signOut()
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/eifandevs/Amby/blob/13b22eabc34a3da4be6d8d051199f3a006a4b490/Amby/Util/Service/LoginService.swift#L53"></span>Force tries should be avoided.<br />
<code>force_try</code> <code>LoginService.swift:53</code></td>
    </tr>
  </tbody>
</table>